### PR TITLE
Better FreeCam.

### PR DESCRIPTION
**Pull request by hrkyr**
just combining fly + freecam to make a better working freecam. This may not be used a lot because hive patched it but a good way to see your surroundings without it spoofing you into creative mode when you disable it and other bugs.

### CODE REVIEW

**Comment by hrkyr** [comment]
we love horion | lets make it bypass hive !!!!!!!

**Comment by Intoprelised** [comment]
true

**Comment by GodGamer029** [comment]
Why are you setting the timer to 20 tps three times in that module? It doesn't do anything

**Comment by GodGamer029** [comment]
When that is resolved I'm willing to merge, but we should get the debug camera working at some point in the future

**Comment by hrkyr** [comment]
> When that is resolved I'm willing to merge, but we should get the debug camera working at some point in the future

fixed. sorry idk why I do that either.

**Comment by hrkyr** [comment]
it failed? ill try again..

**Comment by hrkyr** [comment]
fixed

File: Horion/Module/Modules/Freecam.cpp [hunk 1]
```diff
@@ -53,6 +55,6 @@ void Freecam::onDisable() {
 	if (g_Data.getLocalPlayer() != nullptr) {
 		g_Data.getLocalPlayer()->setPos(oldPos);
 		if (g_Data.getLocalPlayer()->gamemode != 1)
```
Comment (OldGreggo5081) [inline Horion/Module/Modules/Freecam.cpp]: i dont think theres need for "if (g_Data.getLocalPlayer()->gamemode != 1)" or the contents of it

Comment (hrkyr) [reply Horion/Module/Modules/Freecam.cpp]: well it was in the original freecam so I didn't bother to take it out or tamper with it

Comment (OldGreggo5081) [reply Horion/Module/Modules/Freecam.cpp]: well theres no need for it as the module doesnt change the gamemode

**Comment by hrkyr** [comment]
alright again

**Comment by OldGreggo5081** [comment]
looks good to me

**Comment by richardletshacks** [comment]
i'll just merge don't care if it doesn't work :trollface:

**Pull request merged**